sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -223,12 +223,6 @@ trait FileFormat {
    */
   def fileConstantMetadataExtractors: Map[String, PartitionedFile => Any] =
     FileFormat.BASE_METADATA_EXTRACTORS
-
-  /**
-   * Returns whether the file format supports filter push down
-   * for non utf8 binary collated columns.
-   */
-  def supportsCollationPushDown: Boolean = false
```

Comment:
I'm not too familiar with this stuff, but I'm interested in the general picture

we remove this flag, and only allow `deterministic` filters through?

Reply:
We used this flag to determine whether to push filters with collations to the data source, but since we added the logic for widening this no longer makes sense. I just reverted to the old logic which just checked for determinism of the filters